### DISCUSSION TITLE
Proposal for documentation tooling enhancements

### DISCUSSION THREAD

**Discussion by HorizonNet** [unitycatalog/unitycatalog#572] [Ideas]
Denny recently opened a bunch of documentation tickets. When going through them I noticed that there’s not only a difference between the documentation from a layout perspective, but also from an editing one. On [another ticket](https://github.com/unitycatalog/unitycatalog/issues/564) a link to an external page was broken.

That’s why I would like to propose the introduction of two tools into the documentation editing process:

* [markdownlint](https://github.com/DavidAnson/markdownlint/) - This should help to have a common style for the Markdown files.
* [markdown-link-check](https://github.com/tcort/markdown-link-check) - To ensure that there are no broken links in the documentation, especially the external ones we’re not controlling.

Integration could happen in different ways, but probably the easiest would be to have them in a GitHub workflow.
Any thoughts?

**Comment by dennyglee** [comment]
I like the approach @HorizonNet  - by any chance have you played with the integration with VSCode or Sublime for these two tools?  I'd love to get feedback if there are other tools that we should consider as well, eh?!  But with my _very limited_ understanding of the tools, seems like this is a good idea to at least try, eh?!

**Reply by HorizonNet** [reply]
[markdownlint in VS Code](https://marketplace.visualstudio.com/items?itemName=DavidAnson.vscode-markdownlint) works pretty well. I'm using it for quite a while now. As I'm not using Sublime, can't tell how it looks there. For markdown-link-check I didn't find an integration in VS Code. I used it on the command line for the ticket mentioned above.

**Reply by dennyglee** [reply]
Thanks @HorizonNet  - unless there are any issues, I'm inclined to agreed with you to introduce these two tools.  Could you propose a PR/doc change/workflow change to incorporate these tools?

**Reply by HorizonNet** [reply]
@dennyglee Will do. Just one question before I start: Should the changes necessary to make the docs comply with these tools be included in this PR or should I break it down into reviewable pieces, f.e. 3 pages per PR and 1 PR for the workflow (to also check what's left open), which gets merged last?

**Reply by dennyglee** [reply]
Thanks @HorizonNet  - I would suggest breaking this down in reviewable pieces so folks can have a chance to grok what's going on prior to us applying the workflow, eh?!

**Reply by HorizonNet** [reply]
@dennyglee A first change is proposed in #648. Please let me know if it's too much at once. Will wait for additional ones until this one has been reviewed.

**Reply by dennyglee** [reply]
Thanks @HorizonNet - I added a quick comment to #648  - could you please review and address it when you get the chance?  Much appreciated!

**Reply by HorizonNet** [reply]
@dennyglee Done. As I saw that you use a different indentation as I do. Did you already think about Editorconfig to have the IDE behave the same way for every dev?

**Reply by HorizonNet** [reply]
@dennyglee I just noticed that we have two pages in the documentation, which are not included in the published version (`integrations/unity-catalog-celerdata.md` and `server/google-auth.md`). Is this on purpose or is it just missing in the MkDocs configuration?

**Reply by dennyglee** [reply]
Oh, I thought we had fixed the `integrations/unity-catalog-celerdata.md` in a different PR.  So yeah, let's just get that fixed if it isn't already.  The `server/google-auth.md` is on purpose because we will want to add other examples (e.g. Okta) but we're not (nor should we be) the primary resource for how to do this.

**Reply by HorizonNet** [reply]
I looked through the open PRs for the CelerData one and wasn't able to find something adding it to the nav. Let me fix it.

**Reply by dennyglee** [reply]
Thanks @HorizonNet - much appreciated!
